File: 1513-number-of-substrings-with-only-1s/1513-number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        MOD = 10**9 + 7
        cur = 0
        ans = 0
        for c in s:
            if c == '1':
                cur = (cur + 1) % MOD
            else:
                cur = 0
            ans = (ans + cur) % MOD
        return ans

File: 1513-number-of-substrings-with-only-1s/1513-number-of-substrings-with-only-1s.java
class Solution {
    public int numSub(String s) {
        final int MOD = 1000000007;
        long cur = 0, ans = 0;
        for(char c: s.toCharArray()){
            if(c == '1') cur = (cur + 1) % MOD;
            else cur = 0;
            ans = (ans + cur) % MOD;
        }
        return (int)ans;
    }
}